### PR TITLE
IntegerField & DoubleField set default value back when language is changed

### PR DESCRIPTION
**Pull request by boutinb**
Solves also the layout problem

Solves jasp-stats/jasp-test-release#670

### CODE REVIEW

File: JASP-Desktop/widgets/boundqmltextinput.cpp [hunk 1]
```diff
@@ -274,6 +274,19 @@ bool BoundQMLTextInput::isJsonValid(const Json::Value &optionValue)
 	return valid;
 }
 
+void BoundQMLTextInput::setUp()
+{
+	if (form())
+		// For unknown reason, when the language is changed, QML reset the default value.
```
Comment (JorisGoosen) [inline JASP-Desktop/widgets/boundqmltextinput.cpp:280]: QML resets all bindings on language changed, this is not unknown, it is an implementation detail. The weird thing is that the reverting to the default value (to zero anyway) only happens with these few elements...

**Comment by JorisGoosen** [comment]
Well, your fix works... But, I still think there must be a different explanation why the values get reset for these two fields and not the percentage field next to it. I hope you do not mind that I will first take a look at that before I merge this?

**Comment by JorisGoosen** [comment]
Ok, I guess your solution is good enough :)

But you know what would've been better here? A clean separation of view and data. So not storing defaultValue and value in QML but in the boundqmltextinput. And putting the logic there as well.

Instead of what we have now...

**Comment by boutinb** [comment]
What strange is, is that this problem depends on the defaultValue I gave in the DoubleField or  IntegerField... with a 0 or 10, I got the error, but not with 20... that's why I gave up to try to understand what was the real problem. But your suggestion to set the defaultValue from the model is anyway a good idea.

**Comment by JorisGoosen** [comment]
Aah yeah, I thought it was weird as well that the `PercentField`'s default value works, despite it also secretly being a `DoubleField`...

**Pull request merged**